### PR TITLE
Throw NotAuthorizedException on 401 response

### PR DESCRIPTION
**Pull request by diego1auto**

### CODE REVIEW

**Review by daydiff — APPROVED** [review]

**Review by aivus — APPROVED** [review]

**Comment by aivus** [comment]
@diego1auto ping me when you want to merge it

**Review by justenj — APPROVED** [review]

File: Service/Response/ResponseTransformer.php
Comment (aviator-ua) [inline Service/Response/ResponseTransformer.php:1]: NIT. Let's start thinking about some HttpExceptionFactory in the future

File: Service/Response/ResponseTransformer.php [hunk 1]
```diff
@@ -89,6 +90,13 @@ public function transform(ResponseInterface $response, ServiceRequestInterface $
                 throw new NotFoundException($errorDTO, $errorDTO->getStatus(), $errorDTO->getMessage());
             }
 
+            if ($response->getStatusCode() === Response::HTTP_UNAUTHORIZED) {
```
[the remark this answers is not in the document]
Comment (diego1auto) [reply Service/Response/ResponseTransformer.php:93]: There is only 2 test flows, successful and general fail to ensure a ResponseException is thrown

**Review by milan-miscevic — APPROVED** [review]

File: Service/Response/ResponseTransformer.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (aviator-ua) [reply Service/Response/ResponseTransformer.php:93]: Could you add tests for this new functionality?

[the remark this answers is not in the document]
Comment (diego1auto) [reply Service/Response/ResponseTransformer.php:93]: reworked the unit test to cover all cases

**Pull request merged**